tests/scryer/shared_library_tests.rs

```diff
@@ -0,0 +1,141 @@
+        let query = CString::new("X=Y.").unwrap();
+        let machine_ptr: *mut Machine = machine_new();
+        // Check if pointer is not null
+        assert!(!machine_ptr.is_null());
+        unsafe {
+            consult_module_string(&mut *machine_ptr, module_name.as_ptr(), program.as_ptr());
+        }
+        let query_state = unsafe { run_query_iter(&mut *machine_ptr, query.as_ptr()) };
+
+        // should be X=Y not Y=X, see https://github.com/mthom/scryer-prolog/pull/2465#issuecomment-2294961856
+        // expected fix with https://github.com/mthom/scryer-prolog/pull/2475
+        let expected_results = [
+            "{\"result\":[{\"Y\":\"X\"}],\"status\":\"ok\"}", // should be:
+                                                              // "{\"result\":[{\"X\":\"Y\"}],\"status\":\"ok\"}"
+        ];
+
+        let query_state_ref = unsafe { &mut *query_state };
+        for expected in &expected_results {
+            let result_ptr = run_query_next(&mut *query_state_ref);
+            let result_char = unsafe { CStr::from_ptr(result_ptr) };
+            let result_s = result_char.to_str().unwrap();
+            let result_obj = serde_json::from_str::<serde_json::Value>(result_s).expect("Bad JSON");
+            let expected_obj =
+                serde_json::from_str::<serde_json::Value>(expected).expect("Bad JSON");
+            println!("{result_s:?}");
+            assert_eq!(result_obj, expected_obj);
+            unsafe {
+                free_c_string(result_ptr);
+            }
+        }
+
+        unsafe { query_state_free(query_state) }
+        // causes an error if uncommented
+        // unsafe { machine_free(machine_ptr) };
+    }
+
+    #[test]
+    fn test_scryer_run_query_true_members() {
+        let program =
+            CString::new(":- use_module(library(lists)). :- use_module(library(dif)).").unwrap();
+        let module_name = CString::new("facts").unwrap();
+        let query =
+            CString::new("member(X, [a,\"a\",f(a),\"f(a)\", true, \"true\", false, \"false\"]).")
+                .unwrap();
+        let machine_ptr: *mut Machine = machine_new();
+        unsafe {
+            consult_module_string(&mut *machine_ptr, module_name.as_ptr(), program.as_ptr());
+        }
+        let query_state = unsafe { run_query_iter(&mut *machine_ptr, query.as_ptr()) };
+
+        let expected_results = [
+            "{\"result\":[{\"X\":\"a\"}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\":\"\\\"a\\\"\"}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\":\"f(a)\"}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\":\"\\\"f(a)\\\"\"}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\": true}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\":\"\\\"true\\\"\"}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\": false}],\"status\":\"ok\"}",
+            "{\"result\":[{\"X\":\"\\\"false\\\"\"}],\"status\":\"ok\"}",
+        ];
```

Comment:
To make strings containing a lot of escape sequences more readable you could use [raw string literals](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals). For example:

```rust
r#"{"result":[{"X":"a"}],"status":"ok"}"#
```